### DISCUSSION TITLE
How to apply custom opacity to background colors in Tailwind CSS without creating multiple classes?

### DISCUSSION THREAD

**Discussion by MonaAghili** [tailwindlabs/tailwindcss#14215] [Help]
I am working on a design system where users can apply different opacity levels to background colors. Since we cannot use dynamic classes in our codebase, we need to consider how to implement this in Tailwind CSS v3.

https://tailwindcss.com/docs/background-color#changing-the-opacity
With the current method in Tailwind v3 requires creating multiple classes for each opacity level of a background color, which is not an efficient solution for us.

I am considering using the `bg-opacity-` class from Tailwind v2, which allows us to set opacity values without needing to create multiple classes. My question is: Is it safe to use this class, or is there a risk that it could be deprecated in the future?

**Comment by wongjn** [comment]
I recall that I've seen the maintainers mentioned that it is deprecated and hence not in the documentation for v3. If you don't want to have multiple classes for opacity levels, you could consider using CSS variables instead.

**Reply by MonaAghili** [reply]
Thanks

**Comment by adamwathan** [comment]
Hey! I'd do it like this:

```html
<div class="bg-blue-500/[--bg-opacity]" style="--bg-opacity: {dynamicValue}">
  ...
</div>
```

**Reply by MonaAghili** [reply]
Thank you, I will try this out.
